Documentation/ApiOverview/Events/Events/Linkvalidator/BeforeRecordIsAnalyzedEvent.rst

```diff
@@ -10,6 +10,144 @@ BeforeRecordIsAnalyzedEvent
+                  event: TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent
+
+For the implementation we need the :php:`BrokenLinkRepository` to register
+additional link errors and the :php:`SoftReferenceParserFactory` so we can
+automatically parse for links. These two classes have to be injected via
+:ref:`dependeny injection <Dependency-Injection>`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    use TYPO3\CMS\Core\DataHandling\SoftReference\SoftReferenceParserFactory;
+    use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+    use TYPO3\CMS\Linkvalidator\Repository\BrokenLinkRepository;
+
+    class BeforeRecordIsAnalyzedEventListener
+    {
+        protected BrokenLinkRepository $brokenLinkRepository;
+        protected SoftReferenceParserFactory $softReferenceParserFactory;
+
+        public function __construct(
+            BrokenLinkRepository $brokenLinkRepository,
+            SoftReferenceParserFactory $softReferenceParserFactory
+        ) {
+            $this->brokenLinkRepository = $brokenLinkRepository;
+            $this->softReferenceParserFactory = $softReferenceParserFactory;
+        }
+    }
+
+In this example we only check the field :sql:`bodytext` of new tables:
+
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    public function handleEvent(BeforeRecordIsAnalyzedEvent $event): void
+    {
+        $table = $event->getTableName();
+        $forbiddenDomain = 'example.org';
+        if ($table == 'tx_news_domain_model_news') {
+            $results = $event->getResults();
+            $newsItem = $event->getRecord();
+            // a quick check if the forbidden domain is contained at all
+            if (strpos((string) $newsItem['bodytext'], $forbiddenDomain) !== false) {
+                $field = 'bodytext';
+                $conf = $GLOBALS['TCA']['tx_news_domain_model_news']['columns']['bodytext']['config'];
+                // parse the field
+                foreach ($this->softReferenceParserFactory->getParsersBySoftRefParserList($conf['softref'], ['subst']) as $softReferenceParser) {
+                    $parserResult = $softReferenceParser->parse($table, $field, $newsItem['uid'], $newsItem['bodytext']);
+                    if (!$parserResult->hasMatched()) {
+                        continue;
+                    }
+                    // act on each link found
+                    foreach ($parserResult->getMatchedElements() as $matchedElement) {
+                        // if it contains the forbidden domain, subdomains would also be allerted here
+                        if ($matchedElement['subst'] && $matchedElement['subst']['tokenValue']
+                            && strpos($matchedElement['subst']['tokenValue'] . '', $forbiddenDomain) !== false) {
```

Comment:
Use str_contains instead of strpos, as polyfill is in place for v11.

Reply:
I though this only works starting with PHP 8.0 and v11 still supports 7.4?

Reply:
Polyfill is in place in v11, so it can be used safely: https://github.com/TYPO3/typo3/blob/11.5/composer.json#L88